test/test_polygon.py

```diff
@@ -124,20 +125,205 @@ def test_copy_invalid_args(self):
         po_2 = po.copy()
 
         self.assertEqual(po_2.vertices, [p1, p2, p3, p4])
         self.assertEqual(po_2.vertices, po.vertices)
 
+    def test_center_x(self):
+        """Makes sure changing center x component does change the positions of vertices properly."""
+        vertices = _some_vertices.copy()
+        poly = Polygon(vertices)
+        poly_c_x = poly.c_x
+        poly.c_x = 100.0
+        for vertex in vertices:
+            vertex[0] += 100.0 - poly_c_x
+
+        vertices = [tuple(vertex) for vertex in vertices]
+        self.assertEqual(poly.vertices, vertices)
+
+    def test_center_x__invalid_value(self):
+        """Ensures the function can handle the polygon center x component by invalid data types."""
+        poly = Polygon(_some_vertices.copy())
+        for value in (None, [], "1", (1,), [1, 2, 3]):
+            with self.assertRaises(TypeError):
+                poly.c_x = value
+    
+    def test_center_x__del(self):
+        """Ensures that x component cannot be deleted."""
+        poly = Polygon(_some_vertices.copy())
+        with self.assertRaises(AttributeError):
+            del poly.c_x
+
+    def test_center_y(self):
+        """Makes sure changing center y component does change the positions of vertices properly."""
+        vertices = _some_vertices.copy()
+        poly = Polygon(vertices)
+        poly_c_y = poly.c_y
+        poly.c_y = 100.0
+        for vertex in vertices:
+            vertex[1] += 100.0 - poly_c_y
+
+        vertices = [tuple(vertex) for vertex in vertices]
+        self.assertEqual(poly.vertices, vertices)
+
+    def test_center_y__invalid_value(self):
+        """Ensures the function can handle the polygon center y component by invalid data types."""
+        poly = Polygon(_some_vertices)
+        for value in (None, [], "1", (1,), [1, 2, 3]):
+            with self.assertRaises(TypeError):
+                poly.c_y = value
+
+    def test_center_y__del(self):
+        """Ensures that y component cannot be deleted."""
+        poly = Polygon(_some_vertices)
+        with self.assertRaises(AttributeError):
+            del poly.c_y
+
+    def test_center(self):
+        """Makes sure that setting new center moves the vertices properly."""
+        poly = Polygon(_some_vertices.copy())
+        poly_center = poly.center
+        poly.center = (200.0, 200.0)
+        vertices = _some_vertices.copy()
+        for vertex in vertices:
+            vertex[0] += 200.0 - poly_center[0]
+            vertex[1] += 200.0 - poly_center[1]
+
+        vertices = [tuple(vertex) for vertex in vertices]
+        self.assertEqual(poly.vertices, vertices)
+        self.assertEqual(poly.c_x, 200.0)
+        self.assertEqual(poly.c_y, 200.0)
+
+    def test_center__invalid_value(self):
+        """Ensures the function can handle the polygon center component by invalid data types."""
+        poly = Polygon(_some_vertices.copy())
+        for value in (None, [], "1", (1,), [1, 2, 3]):
```

Comment:
Might be wise to test for a valid type (a tuple of two elements, but one of the elements is wrong, like:
(1, "s"), (None, 3), (2, (3))

Reply:
wow, this makes really sense. Alright let me patch this up.